test/vinyl-luatest/update_optimize_test.lua

```diff
@@ -0,0 +1,405 @@
+local t = require('luatest')
+local g = t.group()
+local fiber = require('fiber')
+local log = require('log')
+
+local space
+local dump_count
+local new_stmt_count
+local old_stmt_count
+local index, index2, index3
+
+g.before_all = function()
+    pcall(log.cfg, {level = 6})
+    box.cfg {
+        listen            = os.getenv("LISTEN"),
+        memtx_memory      = 512 * 1024 * 1024,
+        memtx_max_tuple_size = 4 * 1024 * 1024,
+        vinyl_read_threads = 2,
+        vinyl_write_threads = 3,
+        vinyl_memory = 512 * 1024 * 1024,
+        vinyl_range_size = 1024 * 64,
+        vinyl_page_size = 1024,
+        vinyl_run_count_per_level = 1,
+        vinyl_run_size_ratio = 2,
+        vinyl_cache = 10240, -- 10kB
+        vinyl_max_tuple_size = 1024 * 1024 * 6,
+    }
+
+    require('console').listen(os.getenv('ADMIN'))
+
+    space = box.schema.space.create('test', { engine = 'vinyl' })
```

Comment:
As I mentioned in the previous comment, I don't think it's worth creating one space for all tests - after all they use different indexes anyway and have to create them separately. Please try to move space creation to each test case so that they can be executed in an arbitrary order.

Reply:
Now we create in the `before_each` section.